chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -783,12 +783,16 @@ impl PeerManagerActor {
                 self.state.tier2.broadcast_message(Arc::new(PeerMessage::Block(block)));
                 NetworkResponses::NoResponse
             }
-            NetworkRequests::OptimisticBlock { optimistic_block } => {
-                // TODO(#10584): send this message to all the producers.
-                // Maybe we just need to send this to the next producers.
-                self.state
-                    .tier1
-                    .broadcast_message(Arc::new(PeerMessage::OptimisticBlock(optimistic_block)));
+            NetworkRequests::OptimisticBlock { block_producers, optimistic_block } => {
+                // TODO(saketh): the block_producers are identified by their validator AccountId,
+                // but OptimisticBlock is sent over a direct PeerMessage. Hence we have to perform
+                // a conversion here from AccountId to peer connection. Consider reworking this.
+                let msg = Arc::new(PeerMessage::OptimisticBlock(optimistic_block));
+                for target_account in block_producers {
+                    if let Some(conn) = self.state.get_tier1_proxy_for_account_id(&target_account) {
```

Comment:
Is it possible that we silently drop the message in case we are not able to find the tier1 proxy?

Reply:
Yes. Note that this is not a behavior change: OB was previously being broadcast to T1 peers only. 

If OB were a RoutedMessage instead of a PeerMessage we could use `send_message_to_account` which falls back to routing over T2 if the T1 connection is absent. But that change involves a more difficult migration and we will revisit this in the future.